pkg/openstackconfigversion/git_util.go

```diff
@@ -0,0 +1,221 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package openstackconfigversion
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"regexp"
+	"strings"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	git "github.com/go-git/go-git/v5"
+	config "github.com/go-git/go-git/v5/config"
+	"github.com/go-git/go-git/v5/plumbing/format/diff"
+	"github.com/go-git/go-git/v5/plumbing/transport/ssh"
+	"github.com/go-git/go-git/v5/storage/memory"
+
+	crypto_ssh "golang.org/x/crypto/ssh"
+
+	"github.com/go-logr/logr"
+	ospdirectorv1beta1 "github.com/openstack-k8s-operators/osp-director-operator/api/v1beta1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// Patch is an implementation of fdiff.Patch interface
+type Patch struct {
+	message     string
+	filePatches []diff.FilePatch
+}
+
+// FilePatches .
+func (p Patch) FilePatches() []diff.FilePatch {
+	return p.filePatches
+}
+
+// Message .
+func (p Patch) Message() string {
+	return p.message
+}
+
+// filterPatches internal function which is used to strip out '_server_id', ansible inventory, and config changes from our git diff results
+func filterPatches(filePatches []diff.FilePatch) []diff.FilePatch {
+	var retPatches []diff.FilePatch
+
+	m1 := regexp.MustCompile(`_server_id:`)
+	for _, fp := range filePatches {
+		from, to := fp.Files()
+
+		if len(fp.Chunks()) == 0 {
+			continue
+		}
+		if from == nil {
+			// New File is created.
+			retPatches = append(retPatches, fp)
+		} else if to == nil {
+			// File is deleted.
+			retPatches = append(retPatches, fp)
+
+		} else if from.Path() == to.Path() {
+
+			if strings.HasSuffix("tripleo-ansible/tripleo-ansible-inventory.yaml", from.Path()) || strings.HasSuffix("tripleo-ansible/ansible.cfg", from.Path()) {
+				continue
+			}
+
+			var chunks []diff.Chunk
+			for _, chunk := range fp.Chunks() {
+				chunk.Type()
+				s := chunk.Content()
+				if len(s) == 0 {
+					continue
+				}
+
+				if chunk.Type() == diff.Add {
+
+					if !m1.Match([]byte(s)) {
+						chunks = append(chunks, chunk)
+					}
+				}
+
+			}
+			if len(chunks) > 0 {
+				retPatches = append(retPatches, fp)
+			}
+		} else {
+			// file is renamed
+			retPatches = append(retPatches, fp)
+		}
+	}
+
+	return retPatches
+}
+
+// SyncGit func
+func SyncGit(inst *ospdirectorv1beta1.OpenStackPlaybookGenerator, client client.Client, log logr.Logger) (map[string]ospdirectorv1beta1.OpenStackConfigVersion, error) {
+
+	configVersions := make(map[string]ospdirectorv1beta1.OpenStackConfigVersion)
+
+	// Check if this Job already exists
+	foundSecret := &corev1.Secret{}
+	err := client.Get(context.TODO(), types.NamespacedName{Name: inst.Spec.GitSecret, Namespace: inst.Namespace}, foundSecret)
+	if err != nil {
+		log.Info("GitRepo err")
```

Comment:
this message should probably  be more specific to say that the inst.Spec.GitSecret secret was not found instead of  this generic error message?